<launch>
<!-- Launch file for merged Livox LiDAR (159, 160, 161) and FAST-LIVO2 mapping -->

	<arg name="rviz" default="true" />

	<!-- 启动LiDAR话题合并节点 -->
	<node pkg="fast_livo" type="livox_multi_lidar" name="livox_multi_lidar" output="screen">
		<param name="output_topic" value="/livox/multi_lidar" />
		<param name="input_topic_159" value="/livox/lidar_192_168_1_159" />
		<param name="input_topic_160" value="/livox/lidar_192_168_1_160" />
		<param name="input_topic_161" value="/livox/lidar_192_168_1_161" />
	</node>

	<!-- 加载合并后的LiDAR配置 -->
	<rosparam command="load" file="$(find fast_livo)/config/livox_multi_lidar.yaml" />
	
	<!-- 启动FAST-LIVO2建图节点 -->
	<node pkg="fast_livo" type="fastlivo_mapping" name="laserMapping" output="screen">
		<rosparam file="$(find fast_livo)/config/camera_pinhole.yaml" />
	</node>
	
	<group if="$(arg rviz)">
		<!-- 使用机器人跟随视角的RViz配置 -->
		<node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find fast_livo)/rviz_cfg/livox_multi_lidar.rviz" />
	</group>
	
	<node pkg="image_transport" type="republish" name="republish" args="compressed in:=/left_camera/image raw out:=/left_camera/image" output="screen" respawn="true"/>

launch-prefix="gdb -ex run --args"
launch-prefix="valgrind --leak-check=full"
</launch>